server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -136,6 +147,20 @@ public int getCurrentQueueSize() {
         return getQueue().size();
     }
 
+    public double getPercentPoolUtilizationEWMA() {
+        if (trackUtilizationEWMA == false) {
+            return 0;
+        }
+        return this.percentPoolUtilizationEWMA.getAverage();
+    }
+
+    public double getQueuedTaskLatencyMillisEWMA() {
+        if (trackQueueLatencyEWMA == false) {
+            return 0;
+        }
+        return queueLatencyMillisEWMA.getAverage();
+    }
```

Comment:
Some service needs to call these methods. Is it going to be `AverageWriteLoadSampler` or is it still in discussion?